tests/test_router.py

```diff
@@ -53,6 +55,45 @@ def test_other(self):
         from django.contrib.auth import get_user_model
         self.assertEqual(self.sut.db_for_read(model=get_user_model()), None)
 
+    def test_router_gets_filter_hints(self):
+        TestModel.objects.create(user_pk=self.user.pk)
+
+        lookups_to_find = {'exact_lookups': {'user_pk': self.user.pk}}
+
+        with patch.object(ShardedRouter, 'db_for_read') as read_route_function:
+            read_route_function.return_value = 'app_shard_001'
+            test_models = list(TestModel.objects.filter(user_pk=self.user.pk))
+            self.assertIn(lookups_to_find, read_route_function.call_args)
+
+    def test_queryset_router_filter_returns_existing_objects(self):
```

Comment:
something like `test_queryset_filter_kwargs__passed_to_router__returns_existing_objects` would make it easier to understand from the title alone.